hash_practice/exercises.py

```diff
@@ -22,8 +54,25 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
+        Time Complexity: o(n^2)
+        Space Complexity: O(n)
+#     """
+from collections import defaultdict     
+def validSudoku(board):

```

Comment:
This actually looks like a working implementation! However because your function is called `validSudoku` in camel-case instead of `valid_sudoku` as our automated tests expect, the tests are failing.

This problem is optional, but if you would like to pass the tests when you make that fix to get the anagram tests passing, just change the name here.